src/screens/Documents/View/index.js

```diff
@@ -112,42 +130,182 @@ class DocumentsViewScreen extends NavigatorComponent<Props & DocumentsState & Ac
+      Alert.alert(
+        'Signature', `${result.value}`,
+        [
+          {
+            text: 'SHARE',
+            // onPress: () => this.props.navigator.dismissModal(),
+          },
+          {
+            text: 'COPY',
+            onPress: () => this.writeToClipboard(result.value),
+          },
+        ],
+      );
+    } else if (result && result.name === i18n.t('screens.documentView.dochash') && result.value) {
+      Alert.alert(
+        'Document Hash', `${result.value}`,
+        [
+          {
+            text: 'SHARE',
+            //  onPress: () => this.props.navigator.dismissModal(),
+          },
+          {
+            text: 'COPY',
+            onPress: () => this.writeToClipboard(result.value),
+          },
+        ],
+      );
+    } else {
+      console.log('Here');
+    }
+  };
+
+  writeToClipboard = async (value) => {
+    this.setState({ text: value });
+    await Clipboard.setString(this.state.text);
+  };
+
+  txModalClose() {
+    this.setState({ visibleTxModal: false });
+  }
+
   render() {
     const document = getOpenedDocument(this.props);
-    if (document == null) return (<View />);
+    if (document == null) return <View />;
+    const { isUploading } = this.props;
+    const documentDetail = [
+      {
+        name: i18n.t('screens.documentView.registered'),
+        value: document.registered,
+      },
+      {
+        name: i18n.t('screens.documentView.txhash'),
+        value: document.tx_hash,
+      },
+      {
+        name: i18n.t('screens.documentView.signature'),
+        value: document.signature,
+      },
+      {
+        name: i18n.t('screens.documentView.dochash'),
+        value: document.doc_hash,
+      },
+    ];
 
     return (
       <View style={styles.screenContainer}>
         <View style={styles.previewContainer}>
-          <Image source={imageSource(this.content, document.mimeType)} style={styles.preview} resizeMode='contain' />
+          <Image
+            source={imageSource(this.content, document.mimeType)}
+            style={styles.preview}
+            resizeMode='contain'
+          />
         </View>
         <View style={styles.metadataContainer}>
-          <Text style={styles.headline}>
-            {document.name}
-          </Text>
-          <Text style={styles.footnote}>
-            {document.description}
-          </Text>
+          <ScrollView>
+            <Text style={styles.headline}>{document.name}</Text>
+            <Text style={styles.footnote}>{document.description}</Text>
+            {!isUploading && (
+              <Button
+                enabled
+                style={styles.actionButton}
+                title={i18n
+                  .t('screens.documentView.submitdocument')
+                  .toUpperCase()}
+                onPress={this.onDocumentSubmit}
+                styleTitle={styles.settingsText}
+              />
+            )}
+            {isUploading && document.status && (
```

Comment:
Please use `isUploading === true` instead. Please use strict check for `document.status`

Reply:
Done